app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,47 @@
+class VotesController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    @vote = Vote.new(choice: params[:vote][:choice])
+    @vote.user = current_user
+    @vote.review_id = params[:review_id]
+    if @vote.save
+      redirect_to @vote.review.weather, notice: "Thank you for voting."
+    else
+      redirect_to @vote.review.weather,
+        notice: "You have already voted on that review"
```

Comment:
If you wanted, you could set the flash before redirecting: `flash[:notice] = "..."`